### PR TITLE
Added support for HypeTrain Event Subscription via WebHook

### PR DESCRIPTION
**Pull request by HoodyBooti**
Type: Feature
Fixes: #174

Adds support to subscribe to the HypeTrain Event

### CODE REVIEW

File: packages/twitch/src/API/Helix/WebHooks/HelixWebHooksApi.ts [hunk 1]
```diff
@@ -142,6 +142,30 @@ export class HelixWebHooksApi extends BaseApi {
...
+	 *
+	 * @param broadcasterId The broadcaster / channel for which to get notifications about hypetrain events.
+	 * @param options
+	 */
+	async subscribeToHypeTrainEvents(broadcasterId: UserIdResolvable, options: HelixWebHookHubRequestOptions) {
+		return this._sendHypeTrainEventHubRequest('subscribe', broadcasterId, options);
+	}
+
+	/**
+	 * Unsubscribes from events representing a user following other users.
```
Comment (d-fischer) [inline packages/twitch/src/API/Helix/WebHooks/HelixWebHooksApi.ts:158]: This doc block seems to suffer from copypasta ;)

File: packages/twitch/src/API/Helix/HypeTrain/HypeTrainEvent.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+/** @private */
+export interface HypeTrainEventData {
+	eventId: string;
+	broadcasterId: string;
+	level: string;
+	startedAt: string;
+	expiresAt: string;
+	total: string;
+	lastContribution: HypeTrainContributor;
```
Comment (d-fischer) [inline packages/twitch/src/API/Helix/HypeTrain/HypeTrainEvent.ts:13]: Did you mean `HypeTrainContributorData`? You're also gonna need to properly convert the data to class instances in the getters.

**Review by d-fischer — CHANGES_REQUESTED** [review]

File: packages/twitch/src/API/Helix/HypeTrain/HypeTrainEvent.ts [hunk 3]
```diff
@@ -24,6 +24,11 @@ export class HypeTrainEvent {
 	/** @private */
 	constructor(/** @private */ protected _data: HypeTrainEventData, client: ApiClient) {
 		this._client = client;
+		this._data.lastContribution = new HypeTrainContributor(_data.lastContribution);
```
Comment (d-fischer) [inline packages/twitch/src/API/Helix/HypeTrain/HypeTrainEvent.ts:27]: Please don't overwrite raw data with class instances. Instead, convert in the getters, as I said.

**Review by d-fischer — CHANGES_REQUESTED** [review]

**Comment by d-fischer** [comment]
Before I can merge this (regardless of any other changes necessary), please rebase away the file deletions so proper change attribution is kept.

File: packages/twitch/src/API/Helix/HypeTrain/HypeTrainContributor.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import { Enumerable } from '@d-fischer/shared-utils';
+import { ApiClient } from '../../../ApiClient';
+import { UserIdResolvable } from '../../../Toolkit/UserTools';
+
+/**
+ * The type of a hypetrain contribution.
+ */
+export enum HypeTrainContributionType {
+	/**
+	 * A Twitch Partner.
```
Comment (billygerhard) [inline packages/twitch/src/API/Helix/HypeTrain/HypeTrainContributor.ts:10]: This comment doesn't make sense. Was this copy pasta from something else?

**Comment by d-fischer** [comment]
Was this even tested at all? While rebasing and fixing up the problems that were already found, I found more copypasta, both in comments and in actual code, making it impossible to work. Also, the data interface definitions aren't close to the actual data from Twitch.

**Review by Technikkeller — APPROVED** [review]
LGTM

**Review by d-fischer — APPROVED** [review]

**Pull request merged**